internal/backend/temporalclient/client.go

```diff
@@ -76,11 +81,19 @@ func New(cfg *Config, z *zap.Logger) (Client, error) {
 	}, nil
 }
 
-func (c *impl) startDevServer(ctx context.Context, cfg *Config, opts client.Options) error {
-	cfg.DevServer.ClientOptions = &opts
-
+func (c *impl) startDevServer(ctx context.Context) error {
 	var err error
-	if c.srv, err = testsuite.StartDevServer(ctx, cfg.DevServer); err != nil {
+	logPath := path.Join(xdg.DataHomeDir(), "temporal_dev.log")
+	c.logFile, err = os.OpenFile(logPath, os.O_CREATE|os.O_WRONLY, 0o644)
```

Comment:
You are overwriting the log file every time, is this intentional?
Also, why not use `os.Create`?

Reply:
Intentional: yes, (1) to keep the file small over time, (2) the logs of previous runs are rarely useful

Why not `os.Create`: to specify file permissions (644 = `rw-r--r--`) instead of the default (666 = `rw-rw-rw-`)